tests/assets_factory.py

```diff
@@ -48,49 +52,23 @@ def load_predictions(self, path):
     tools.metrics.execute(AccuracyMetric())
 """
 
-DEFAULT_OPENER_FILE = """
+DEFAULT_OPENER_FILE = f"""
 import os
-import shutil
 import numpy as np
 import substratools as tools
 
 class NumpyOpener(tools.Opener):
-    def get_X(self, folders):
-        data = self._get_data(folders)
-        return self._get_X(data)
-
-    def get_y(self, folders):
-        data = self._get_data(folders)
-        return self._get_y(data)
-
-    def fake_X(self, n_samples=None):
-        data = self._fake_data(n_samples)
-        return self._get_X(data)
-
-    def fake_y(self, n_samples=None):
-        data = self._fake_data(n_samples)
-        return self._get_y(data)
-
-    @classmethod
-    def _get_X(cls, data):
-        return data[:, :-1]
-
-    @classmethod
-    def _get_y(cls, data):
-        return data[:, -1:]
-
-    @classmethod
-    def _fake_data(cls, n_samples=None, n_col=3):
-        return np.random.uniform(0, 1, (n_samples, n_col))
-
-    @classmethod
-    def _get_data(cls, folders):
+    def get_data(self, folders):
         paths = []
         for folder in folders:
             paths += [
                 os.path.join(folder, f) for f in os.listdir(folder) if f[-4:] == ".npy"
             ]
-        return np.concatenate([np.load(file, allow_pickle=True) for file in paths], axis=0)
+        data = np.concatenate([np.load(file, allow_pickle=True) for file in paths], axis=0)
+        return (data[:, :-{LINEAR_N_TARGET}], data[:, -{LINEAR_N_TARGET}:])
+
+    def fake_data(self, n_samples=None, n_col={LINEAR_N_COL + LINEAR_N_TARGET}):
+        return np.random.uniform(0, 1, (n_samples, n_col))
```

Comment:
I do not understand this: why do we go from 

```python
x = data[:, :-1]
y = data[:, -1:]
```

to 

`(x, y) = (data[:, :-{LINEAR_N_TARGET}], data[:, -{LINEAR_N_TARGET}:]) `

?

Reply:
Because the generated data uses the LINEAR_N_TARGET constant, 
As we want to split X and y, let's use the LINEAR_N_TARGET constant to do so (it was not the case before, now it is)

Reply:
great, thanks